python/cuml/tests/conftest.py

```diff
@@ -403,6 +405,23 @@ def random_seed(request):
 # =============================================================================
 
 
+class DownloadDataPlugin:
+    """Download data before workers are spawned.
+
+    This avoids downloading data in each worker, which can lead to races.
+    """
+
+    def pytest_configure(self, config):
+        if not hasattr(config, "workerinput"):
+            # We're in the controller process, not a worker. Let's fetch all
+            # the datasets we might use.
```

Comment:
Having to hard-code this list appears extraordinary fragile to me. I assume this is not meant as a long-term solution?

Reply:
Yes and no - I don't like having to hardwire things, at the same time building a system where adding a new fixture for a dataset automatically registers that dataset with the plugin seems like a lot of complexity as well. So for now I think delegating the task of "add your dataset here" to a human is a good compromise, especially as we don't seem to add new datasets a lot (and ideally we'd use generated ones as much as possible?!)

The ideal outcome for me would be finding a way to make the `fetch_20newsgroup` function in scikit-learn safe for concurrent use. Then this problem would go away completely.